Wikimate.php

```diff
@@ -858,3 +858,501 @@ private function findSection($section)
 		return -1;
 	}
 }
+
+
+/**
+ * Models a wiki file that can have its properties retrieved and
+ * its contents downloaded and uploaded.
+ * All properties pertain to the current revision of the file.
+ *
+ * @author  Robert McLeod & Frans P. de Vries
+ * @since   October 2016
+ */
+class WikiFile
+{
+	protected $filename  = null;
+	protected $wikimate  = null;
+	protected $exists    = false;
+	protected $invalid   = false;
+	protected $error     = null;
+	protected $edittoken = null;
+	protected $info      = null;
+
+	/*
+	 *
+	 * Magic methods
```

Comment:
PHP's [OOP terminology](http://www.php.net/manual/en/language.oop5.magic.php) for constructor/destructor, toString, and so on.